src/kernel/include/thd.h

```diff
@@ -577,15 +638,42 @@ thd_switch_update(struct thread *thd, struct pt_regs *regs, int issame)
 	return preempt;
 }
 
+#define THD_GET(op, retval, val) \
+	{ \
+		case op: \
+			*retval = val; \
+			break; \
+	}
+
+#define THD_OP(op, t, retval) \
+	{ \
+		THD_GET(THD_GET_TID, retval, t->tid); \
+		THD_GET(THD_GET_FAULT_REG1, retval, t->fault_regs.ax); \
+		THD_GET(THD_GET_FAULT_REG2, retval, t->fault_regs.bx); \
+		THD_GET(THD_GET_FAULT_REG3, retval, t->fault_regs.cx); \
+		THD_GET(THD_GET_FAULT_REG4, retval, t->fault_regs.dx); \
+		THD_GET(THD_GET_FAULT_REG5, retval, t->fault_regs.cs); \
+		THD_GET(THD_GET_FAULT_REG6, retval, t->fault_regs.ds); \
+		THD_GET(THD_GET_FAULT_REG7, retval, t->fault_regs.es); \
+		THD_GET(THD_GET_FAULT_REG8, retval, t->fault_regs.fs); \
+		THD_GET(THD_GET_FAULT_REG9, retval, t->fault_regs.gs); \
+		THD_GET(THD_GET_FAULT_REG10, retval, t->fault_regs.ss); \
+		THD_GET(THD_GET_FAULT_REG11, retval, t->fault_regs.si); \
+		THD_GET(THD_GET_FAULT_REG12, retval, t->fault_regs.di); \
+		THD_GET(THD_GET_FAULT_REG13, retval, t->fault_regs.ip); \
+		THD_GET(THD_GET_FAULT_REG14, retval, t->fault_regs.sp); \
+		THD_GET(THD_GET_FAULT_REG15, retval, t->fault_regs.bp); \
+		THD_GET(THD_GET_FAULT_REG16, retval, t->fault_regs.flags); \
+		THD_GET(THD_GET_FAULT_REG17, retval, t->fault_regs.orig_ax); \
```

Comment:
you love macros.. don't you? :-P

I think we should stick to the standard and not use macros here.. If you're just getting and returning values its fine but if there is a case in the future to do something more in some "given case" and not all, then you'd have to expand the switch case back.. 

Also I don't think there is any benefit of doing this here.. The number of characters in your "case" macro is less (statement could be inaccurate) than the number of characters in the macro call you have. :-)

Reply:
I think the way we have those lines right now
```'
case THD_GET_XXXX: *retval = thd->xxxxxx; break;
```
is quite terse. I don't know if macros are useful here.
But if we had some configuration (ex: CONFIG_GP_REGS_ONLY kind ) where we'd enable only a subset to be accessible, then this would both help generate less code and get rid of those other register switch cases. But I don't think we'd have such a config or need such.
Also when we have architecture specific "gets" we'd probably move this "introspect" code to `chal` layer and would be per architecture?? I'm not sure if macros are all that useful. 

I'd probably just do the normal thing here for "consistency".